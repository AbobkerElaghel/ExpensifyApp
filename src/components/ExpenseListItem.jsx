import React from "react";
import {Link} from "react-router-dom";
const ExpenseListItem = props => (
    <div>
        <Link to={`/edit/${props.id}`}><h3>{props.description}</h3></Link>
        <p>{props.amount} - {new Date(props.createdAt).toLocaleString()}</p>
    </div>
);
export default ExpenseListItem;