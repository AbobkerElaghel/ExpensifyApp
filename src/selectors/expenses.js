import moment from "moment";

// get visible expenses
export default (expenses, {text, sortBy, startDate, endDate}) => (expenses.filter(expense => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment) : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment) : true;
    const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
    return startDateMatch && endDateMatch && textMatch;
}).sort((a, b) => {
    if(sortBy === "amount"){
        return b.amount - a.amount;
    } else if (sortBy === "date") {
        return b.createdAt - a.createdAt;
    }}));
