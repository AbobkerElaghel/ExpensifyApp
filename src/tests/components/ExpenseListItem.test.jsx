import React from "react";
import ExpenseListItem from "../../components/ExpenseListItem.jsx";
import {shallow} from "enzyme";
import expenses from "../fixtures/expenses";

test('should render Expense list item', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});