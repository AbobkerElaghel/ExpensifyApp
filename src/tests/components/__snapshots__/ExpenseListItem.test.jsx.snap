// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense list item 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    195
     - 
    1/1/1970, 1:00:00 AM
  </p>
</div>
`;
